api_tests/registrations/views/test_registration_resource_list.py

```diff
@@ -0,0 +1,206 @@
+        assert resp.status_code == 410
+
+    @pytest.mark.parametrize('user_role', UserRoles)
+    def test_status_code__withdrawn(self, app, user_role):
+        test_artifact, test_auth, registration = configure_test_preconditions(
+            registration_state=RegStates.WITHDRAWN, user_role=user_role
+        )
+        resp = app.get(make_api_url(registration), auth=test_auth, expect_errors=True)
+        assert resp.status_code == 403 if test_auth else 401
+
+    @pytest.mark.parametrize('registration_state', [RegStates.INITIAL, RegStates.PENDING, RegStates.EMBARGO])
+    @pytest.mark.parametrize('user_role', UserRoles.noncontributor_roles())
+    def test_status_code__vol(self, app, registration_state, user_role):
+        test_artifact, test_auth, registration = configure_test_preconditions(
+            registration_state=registration_state, user_role=user_role
+        )
+        provider = registration.provider
+        provider.reviews_workflow = ModerationWorkflows.NONE.value
+        provider.save()
+
+        vol = PrivateLinkFactory(anonymous=False)
+        vol.nodes.add(registration)
+        resp = app.get(make_api_url(registration, vol_key=vol.key), auth=test_auth, expect_errors=True)
+        assert resp.status_code == 200
+
+
+@pytest.mark.django_db
+class TestRegistrationResourceListGETBehavior:
+
+    @pytest.fixture
+    def admin_user(self):
+        return AuthUserFactory()
+
+    @pytest.fixture
+    def registration(self, admin_user):
+        return RegistrationFactory(creator=admin_user, has_doi=True)
+
+    @pytest.fixture
+    def outcome(self, registration):
+        return Outcome.objects.for_registration(registration, create=True)
+
+    @pytest.fixture
+    def artifact_one(self, outcome):
+        return outcome.artifact_metadata.create(
+            identifier=IdentifierFactory(),
+            artifact_type=ArtifactTypes.DATA,
+            finalized=True
+        )
+
+    @pytest.fixture
+    def artifact_two(self, outcome):
+        return outcome.artifact_metadata.create(
+            identifier=IdentifierFactory(),
+            artifact_type=ArtifactTypes.ANALYTIC_CODE,
+            finalized=True
+        )
+
+    @pytest.fixture
+    def draft_artifact(self, outcome):
+        return outcome.artifact_metadata.create(
+            identifier=IdentifierFactory(),
+            artifact_type=ArtifactTypes.MATERIALS,
+            finalized=False
+        )
+
+    @pytest.fixture
+    def deleted_artifact(self, outcome):
+        return outcome.artifact_metadata.create(
+            identifier=IdentifierFactory(),
+            artifact_type=ArtifactTypes.PAPERS,
+            finalized=True,
+            deleted=timezone.now()
+        )
+
+    def test_visibility(
+        self, app, registration, artifact_one, artifact_two, draft_artifact, deleted_artifact, admin_user
+    ):
+        # Only artifacts with `finalized=True` and `deleted=None` should appear/
```

Comment:
Hardly seems worth changing, but:
```suggestion
        # Only artifacts with `finalized=True` and `deleted=None` should appear.
```